src/transformers/models/byt5/tokenization_byt5.py

```diff
@@ -0,0 +1,268 @@
+        make use of token type ids, therefore a list of zeros is returned.
+
+        Args:
+            token_ids_0 (:obj:`List[int]`):
+                List of IDs.
+            token_ids_1 (:obj:`List[int]`, `optional`):
+                Optional second list of IDs for sequence pairs.
+
+        Returns:
+            :obj:`List[int]`: List of zeros.
+        """
+        eos = [self.eos_token_id]
+
+        if token_ids_1 is None:
+            return len(token_ids_0 + eos) * [0]
+        return len(token_ids_0 + eos + token_ids_1 + eos) * [0]
+
+    def build_inputs_with_special_tokens(
+        self, token_ids_0: List[int], token_ids_1: Optional[List[int]] = None
+    ) -> List[int]:
+        """
+        Build model inputs from a sequence or a pair of sequence for sequence classification tasks by concatenating and
+        adding special tokens. A sequence has the following format:
+
+        - single sequence: ``X </s>``
+        - pair of sequences: ``A </s> B </s>``
+
+        Args:
+            token_ids_0 (:obj:`List[int]`):
+                List of IDs to which the special tokens will be added.
+            token_ids_1 (:obj:`List[int]`, `optional`):
+                Optional second list of IDs for sequence pairs.
+
+        Returns:
+            :obj:`List[int]`: List of `input IDs <../glossary.html#input-ids>`__ with the appropriate special tokens.
+        """
+        token_ids_0 = self._add_eos_if_not_present(token_ids_0)
+        if token_ids_1 is None:
+            return token_ids_0
+        else:
+            token_ids_1 = self._add_eos_if_not_present(token_ids_1)
+            return token_ids_0 + token_ids_1
+
+    def _tokenize(self, text: str) -> List[str]:
+        """Take as input a string and return a list of strings (tokens) for words/sub-words"""
+
+        def _sub_tokenize(sub_text):
+            character_list = list(sub_text)
+            utf_tokens_lists = [list(char.encode("utf-8")) for char in character_list]
+            sub_tokens = [chr(utf_token) for utf_tokens in utf_tokens_lists for utf_token in utf_tokens]
+            return sub_tokens
+
+        # split on special characters
+        pattern = f"({'|'.join(self.special_tokens_encoder.keys())})"
+        sub_texts = list(filter(None, re.split(pattern, text)))
+        tokens = []
+        for sub_text in sub_texts:
+            if sub_text in self.special_tokens_encoder.keys():
+                tokens += [sub_text]
+            else:
+                tokens += _sub_tokenize(sub_text)
+
+        return tokens
+
+    def _convert_token_to_id(self, token):
+        """Converts a token (str) in an id using the vocab."""
+        if token.startswith("<extra_id_"):
+            match = re.match(r"<extra_id_(\d+)>", token)
+            num = int(match.group(1))
+            token_id = self.vocab_size - num - 1
+        elif token in self.special_tokens_encoder:
+            token_id = self.special_tokens_encoder[token]
+        elif len(token) > 1:
```

Comment:
This is only possible for newly added tokens, but required so that additional tokens can be added